.github/workflows/src/arm-auto-signoff/arm-auto-signoff-status.js

```
@@ -201,10 +157,109 @@ export async function getLabelActionImpl({ owner, repo, issue_number, head_sha,
+    repo,
+    run_id: run.id,
+    per_page: PER_PAGE_MAX,
+  });
+  const artifactNames = artifacts.map((a) => a.name);
+
+  core.info(`${wfName} artifactNames: ${JSON.stringify(artifactNames)}`);
+
+  // check for new combined artifact
+  const combinedArtifact = artifactNames.find(name => name.startsWith("arm-auto-signoff-code-results="));
+  if (combinedArtifact) {
+    try {
+      const resultValue = combinedArtifact.substring("arm-auto-signoff-code-results=".length);
+      // Parse key-value format: incrementalTypeSpec-true,isTrivial-false,qualifies-true
+      // Split by comma, then parse each key-value pair
+      const keyValuePairs = resultValue.split(',');
+      
+      // Extract values directly by key name
+      const incrementalTypeSpec = keyValuePairs[0]?.endsWith('-true') ?? false;
+      const isTrivial = keyValuePairs[1]?.endsWith('-true') ?? false;
+      const qualifiesForAutoSignoff = keyValuePairs[2]?.endsWith('-true') ?? false;
+      
+      core.info(`ARM analysis results: incrementalTypeSpec=${incrementalTypeSpec}, isTrivial=${isTrivial}, qualifiesForAutoSignoff=${qualifiesForAutoSignoff}`);
+      
+      if (qualifiesForAutoSignoff) {
+        const labelsToAdd = [];
+        
+        if (incrementalTypeSpec) {
+          labelsToAdd.push("ARMAutoSignedOff-IncrementalTSP");
+        }
+        if (isTrivial) {
+          labelsToAdd.push("ARMAutoSignedOff-Trivial");
+        }
+        
+        const reason = labelsToAdd.join(", ");
+        core.info(`PR qualifies for auto sign-off: ${reason}`);
+        return { shouldAutoSign: true, reason: reason, labelsToAdd: labelsToAdd };
+      } else {
+        core.info("PR does not qualify for auto sign-off based on ARM analysis");
+        return { shouldAutoSign: false, reason: "No qualifying changes detected" };
+      }
+    } catch (error) {
+      core.warning(`Failed to parse combined ARM analysis results: ${error.message}`);
+      // Fall back to legacy artifact checking
```

We shouldn't need any code for "legacy" artifacts.  In general, we don't try to maintain compat between check versions.  PRs are expected to update from main to get latest checks.  And our checks should "fail closed", so if they get into an unexpected cross-version situation, worst case the PR is blocked.

---

Removed